File: 759_div2/B.py
t = int(input())
for x in range(t):
    n = int(input())
    nums = list(map(int, input().split()))
    if len(set(nums)) == 1:
        print(0)
        continue
 
    x = nums[n - 1]
    k = 0
    max_num = 0
    max_index = 0
    for index, num in enumerate(nums):
        if num >= max_num:
            max_num = num
            max_index = index
    
    if max_index == n - 1:
        print(0)
        continue
    second_max_index = 0
    second_max = 0
    for i in range(max_index + 1, len(nums)):
        if nums[i] >= second_max and nums[i] != max_num:
            second_max = nums[i]
            second_max_index = i

    print(len(nums) - second_max_index)

File: 759_div2/B.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int t, n;
    cin >> t;
    for (int x = 0; x < t; x++) {
        cin >> n; 
        vector<int> nums;
        int temp;
        int max_num = 0;
        for (int i = 0; i < n; i++) {
            cin >> temp;
            nums.push_back(temp);
            max_num = max(max_num, temp);
        }
        int q = nums[n - 1];
        int k = 0;
        while (q != max_num) {
            k += 1;
            vector<int> new_arr;
            for (auto num : nums) {
                if (num > q) {
                    new_arr.push_back(num);
                }
            }
            nums.clear();
            for (auto num : new_arr) {
                nums.push_back(num);
            }
            q = new_arr[new_arr.size() - 1];
        }
        cout << k << endl;
    }
    
    return 0;
}
